public/app/features/sandbox/TestStuffPage.tsx

```diff
@@ -14,26 +26,93 @@ export const TestStuffPage = () => {
+          }
+        } else {
+          return {
+            ...state,
+            selected: isSelected ? [] : [item],
+          };
+        }
+      });
+    },
+  });
+
+  const onToggleElementSelection = () => {
+    setElementSelectionState({ ...elementSelectionState, enabled: !elementSelectionState.enabled });
+  };
+
+  const onClearSelection = (evt: React.PointerEvent) => {
+    setElementSelectionState({ ...elementSelectionState, selected: [] });
+  };
 
   return (
     <Page navModel={{ node: node, main: node }}>
-      <Stack>
+      <Stack direction={'column'} gap={2} onPointerUp={onClearSelection}>
         <LinkToBasicApp extensionPointId="grafana/sandbox/testing" />
-        <Button onClick={() => notifyApp.success('Success toast', 'some more text goes here')} variant="primary">
-          Success
-        </Button>
-        <Button
-          onClick={() => notifyApp.warning('Warning toast', 'some more text goes here', 'bogus-trace-99999')}
-          variant="secondary"
-        >
-          Warning
-        </Button>
-        <Button
-          onClick={() => notifyApp.error('Error toast', 'some more text goes here', 'bogus-trace-fdsfdfsfds')}
-          variant="destructive"
-        >
-          Error
-        </Button>
+        <Text variant="h5">Application notifications (toasts) testing</Text>
+        <Stack>
+          <Button onClick={() => notifyApp.success('Success toast', 'some more text goes here')} variant="primary">
+            Success
+          </Button>
+          <Button
+            onClick={() => notifyApp.warning('Warning toast', 'some more text goes here', 'bogus-trace-99999')}
+            variant="secondary"
+          >
+            Warning
+          </Button>
+          <Button
+            onClick={() => notifyApp.error('Error toast', 'some more text goes here', 'bogus-trace-fdsfdfsfds')}
+            variant="destructive"
+          >
+            Error
+          </Button>
+        </Stack>
+        <Stack gap={3} alignItems={'center'}>
+          <Checkbox
+            label="Enable element selection"
+            value={elementSelectionState.enabled}
+            onChange={() => onToggleElementSelection()}
+          />
+          Selected:
+          {elementSelectionState.selected.map((item) => (
+            <span key={item.id}>{item.id}</span>
+          ))}
+        </Stack>
+        <ElementSelectionContext.Provider value={elementSelectionState}>
+          <VizGridLayout>
+            <PanelChrome title={'Panel title 1'} selectionId="panel-1">
+              Hello
+            </PanelChrome>
+            <PanelChrome title={'Panel title 2'} selectionId="panel-2">
+              Hello
+            </PanelChrome>
+            <PanelChrome title={'Panel title 3'} selectionId="panel-3">
+              Hello
+            </PanelChrome>
+            <PanelChrome title={'Panel title 4'} selectionId="panel-4">
+              Hello
+            </PanelChrome>
+          </VizGridLayout>
```

Comment:
Regarding this demo impl, thoughts
* Remove it 
* Move it to a storybook story for ElementSelectionContext 
* Keep it here